src/snowflake/telemetry/_internal/encoder/otlp/proto/common/log_encoder.py

```diff
@@ -2,18 +2,20 @@
 # Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
 #
 
-from logging import LogRecord
 from typing import Sequence
 
+from opentelemetry.exporter.otlp.proto.http._log_exporter import encoder
 from opentelemetry.proto.collector.logs.v1.logs_service_pb2 import ExportLogsServiceRequest
 from opentelemetry.proto.logs.v1.logs_pb2 import LogsData
+from opentelemetry.sdk._logs import LogData
 
 
-def _encode_logs(batch: Sequence[LogRecord]) -> ExportLogsServiceRequest:
-    # TODO fix this no-op implementation of encode_logs
-    return ExportLogsServiceRequest(resource_logs=[])
+def _encode_logs(batch: Sequence[LogData]) -> ExportLogsServiceRequest:
+    # Will no longer rely on _encode_resource_logs after we upgrade to v1.19.0 or later
+    resource_logs = encoder._encode_resource_logs(batch)
+    return ExportLogsServiceRequest(resource_logs=resource_logs)
 
-def serialize_logs_data(batch: Sequence[LogRecord]) -> bytes:
+def serialize_logs_data(batch: Sequence[LogData]) -> bytes:
```

Comment:
Can we avoid exporting this function? Happy to review a separate PR for that.

Reply:
Done, I moved this inside the `_ProtoLogExporter` class. I moved the other serialize_x_data methods also inside their respective exporter classes.